<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">
  <xacro:macro name="ffw_sg2_sensor_system" params="use_sim:=^|false use_fake_hardware:=^|false fake_sensor_commands:=^|false">
    <ros2_control name="ffw_sg2_sensor" type="system" is_async="true">
      <xacro:unless value="$(arg use_sim)">
        <hardware>
          <xacro:if value="${use_fake_hardware}">
            <plugin>fake_components/GenericSystem</plugin>
            <param name="fake_sensor_commands">${fake_sensor_commands}</param>
            <param name="state_following_offset">0.0</param>
          </xacro:if>
          <xacro:unless value="${use_fake_hardware}">
            <plugin>dynamixel_hardware_interface/DynamixelHardware</plugin>
            <param name="port_name">/dev/ttyUSB1</param>
            <param name="baud_rate">4000000</param>
            <param name="error_timeout_ms">500</param>
            <param name="dynamixel_model_folder">/param/dxl_model</param>
            <param name="number_of_joints">0</param>
            <param name="number_of_transmissions">0</param>
            <param name="disable_torque_at_init">true</param>
            <param name="transmission_to_joint_matrix"></param>
            <param name="joint_to_transmission_matrix"></param>
            <param name="dynamixel_state_pub_msg_name">ffw_sensor/dxl_state</param>
            <param name="get_dynamixel_data_srv_name">ffw_sensor/get_dxl_data</param>
            <param name="set_dynamixel_data_srv_name">ffw_sensor/set_dxl_data</param>
            <param name="reboot_dxl_srv_name">ffw_sensor/reboot_dxl</param>
            <param name="set_dxl_torque_srv_name">ffw_sensor/set_dxl_torque</param>
          </xacro:unless>
        </hardware>
      </xacro:unless>

      <sensor name="imu">
        <state_interface name="IMU ACC SCALE"/>
        <state_interface name="IMU GYRO SCALE"/>
        <state_interface name="IMU TIMESTAMP"/>
        <state_interface name="IMU ACC X"/>
        <state_interface name="IMU ACC Y"/>
        <state_interface name="IMU ACC Z"/>
        <state_interface name="IMU GYRO X"/>
        <state_interface name="IMU GYRO Y"/>
        <state_interface name="IMU GYRO Z"/>
      </sensor>

      <gpio name="imu">
        <param name="type">sensor</param>
        <param name="ID">200</param>
        <state_interface name="IMU ACC SCALE"/>
        <state_interface name="IMU GYRO SCALE"/>
        <state_interface name="IMU TIMESTAMP"/>
        <state_interface name="IMU ACC X"/>
        <state_interface name="IMU ACC Y"/>
        <state_interface name="IMU ACC Z"/>
        <state_interface name="IMU GYRO X"/>
        <state_interface name="IMU GYRO Y"/>
        <state_interface name="IMU GYRO Z"/>
      </gpio>
      <gpio name="led">
        <param name="type">controller</param>
        <param name="ID">91</param>
        <!-- <command_interface name="Neopixel_Head_Left_Red"/>
        <command_interface name="Neopixel_Head_Left_Green"/>
        <command_interface name="Neopixel_Head_Left_Blue"/>
        <command_interface name="Neopixel_Head_Left_Brightness"/>
        <command_interface name="Neopixel_Head_Left_Mode"/>
        <command_interface name="Neopixel_Head_Left_Control_Time"/>
        <command_interface name="Neopixel_Head_Left_Led_Count"/>
        <command_interface name="Dummy_1"/>
        <command_interface name="Neopixel_Head_Right_Red"/>
        <command_interface name="Neopixel_Head_Right_Green"/>
        <command_interface name="Neopixel_Head_Right_Blue"/>
        <command_interface name="Neopixel_Head_Right_Brightness"/>
        <command_interface name="Neopixel_Head_Right_Mode"/>
        <command_interface name="Neopixel_Head_Right_Control_Time"/>
        <command_interface name="Neopixel_Head_Right_Led_Count"/> -->
      </gpio>

    </ros2_control>
  </xacro:macro>
</robot>
